src/compiler/moduleNameResolver.ts

```diff
@@ -1411,13 +1430,13 @@ export function resolveModuleName(moduleName: string, containingFile: string, co
                 result = nodeNextModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference, resolutionMode);
                 break;
             case ModuleResolutionKind.Node10:
-                result = nodeModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference);
+                result = nodeModuleNameResolver(moduleName, containingFile, compilerOptions, host, cache, redirectedReference, resolutionMode ? getConditions(compilerOptions, resolutionMode === ModuleKind.ESNext) : undefined);

```

Comment:
Do we want to consider pushing this conditions lookup out to the caller? Or is it too useful a shorthand to take a `resolutionMode` directly on `resolveModuleName` as meaning a lookup type override for node10/bundler? I'm just thinking that if we push it out to the caller, we can pass the `conditions` argument through to all resolvers (and not just the `node10` and `bundler` ones), which'd be nice for API consumers of the resolver who are running funky custom import/import condition structures, but might be bad for our existing resolution cache APIs (namely, they'd have to go from being mode-aware to being condition-aware). It's probably a good compromise to leave it as-is, messing with the resolution cache's API is a lot of surface area to change just to make an edgecase API use nicer.